### PR TITLE
Add lsp-ivy and helm-lsp when :tools lsp

### PR DESCRIPTION
**Pull request by gagbo**
As described in #2160 

- Not sure if my `:init` block is correct (namely the `after lsp`)
- Not sure if using `doom-leader-code-map` is the way to go
- I think all those settings belong to the completion modules though

### CODE REVIEW

File: modules/completion/helm/config.el [hunk 1]
```diff
@@ -182,3 +182,13 @@ be negative.")
         (lambda (buf &optional _resume) (pop-to-buffer buf)))
   (global-set-key [remap swiper] #'swiper-helm)
   (add-to-list 'swiper-font-lock-exclude #'+doom-dashboard-mode nil #'eq))
+
+(use-package! helm-lsp
+  :when (featurep! :tools lsp)
+  :commands (helm-lsp-workspace-symbol helm-lsp-global-workspace-symbol)
```
Comment (hlissner) [inline modules/completion/helm/config.el]: Remove the parentheses around the :commands list. I omit them when the list is short.

File: modules/completion/ivy/config.el [hunk 2]
```diff
@@ -368,6 +368,15 @@ evil-ex-specific constructs, so we disable it solely in evil-ex."
   (setq prescient-save-file (concat doom-cache-dir "prescient-save.el"))
   (prescient-persist-mode +1))
 
+(use-package! lsp-ivy
+  :when (featurep! :tools lsp)
+  :commands (lsp-ivy-workspace-symbol lsp-ivy-global-workspace-symbol)
```
Comment (hlissner) [inline modules/completion/ivy/config.el]: Same as above.

File: modules/completion/helm/config.el (hunk 1 above)
Comment (gagbo) [reply modules/completion/helm/config.el]: Will do

File: modules/completion/ivy/config.el (hunk 2 above)
Comment (gagbo) [reply modules/completion/ivy/config.el]: Same as above

**Comment by gagbo** [comment]
>     1. Please move these ivy/helm packages to the `:tools lsp` module. In general, ivy/helm plugins should be added to the module they extend.

Ok, so I just check for `(featurep! :completion xxxx)` in `:tools lsp` module ?

>     2. In general, do not modify leader keybinds from modules. Modify them from `modules/config/default/+{evil,emacs}-bindings.el`.

Yes, when I pushed it, I remember all the `:when` in `map!`
 
>     3. This is a lot like imenu on `SPC s i`. Perhaps it should be on `SPC s {j,J}` instead, or perhaps remap `SPC s i` when lsp-mode is enabled?

I didn't know about `SPC s i`. I thought about `SPC c` prefix because a few LSP related commands (rename and format iirc) are already there, and "**s**earch **j**ump" feels less mnemonic than "**c**ode **j**ump"

**Comment by hlissner** [comment]
> Ok, so I just check for (featurep! :completion xxxx) in :tools lsp module ?

Yes.

> I didn't know about SPC s i. I thought about SPC c prefix because a few LSP related commands (rename and format iirc) are already there, and "search jump" feels less mnemonic than "code jump"

True. I was thinking of `SPC s s` for search symbols, but it's already taken by swiper. Perhaps I'll replace `SPC s b` with swiper-isearch so that this could take `SPC s s`?

**Comment by gagbo** [comment]
>     3. This is a lot like imenu on `SPC s i`. Perhaps it should be on `SPC s {j,J}` instead, or perhaps remap `SPC s i` when lsp-mode is enabled?

One (big) difference between the 2 is that imenu works in the current buffer, whereas those lsp completions work on all symbols in the workspace

**Comment by gagbo** [comment]
`doom-search-map j` is already taken to search the jumplist, so I think `c j/J` is better at least for now

**Comment by hlissner** [comment]
Looks good to me. Thanks again for the help!

**Pull request merged**